### PR TITLE
add support for recursive refs

### PR DESCRIPTION
**Pull request by sprsquish**

### CODE REVIEW

File: packages/dashboard-base/src/dataset/special-types.js [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
+    const native = getKeyByValue(v.Native, obj)
+    if (native !== undefined)
+      return `v.Native.${native}`
+  }
+
+  var args = [`"${obj.id}"`]
+  if (obj.class !== undefined) args.push(renderRef(obj.class))
+  if (obj.database !== undefined) args.push(renderRef(obj.database))
+  args = args.join(', ')
+  return `new v.Ref(${args})`
```
Comment (erickpintor) [inline packages/dashboard-base/src/dataset/special-types.js]: It's usually hard for new users to understand the difference between an expression constructor and a value constructor. We've been trying to hide that from users by always referring to the expression constructors (e.g: `q.Ref`). Is it possible to render the recursive ref in a way that the user doesn't need to see the internal `v.Ref`?

**Comment by sprsquish** [comment]
This is dependent on the release of this fix: https://github.com/fauna/faunadb-js/pull/155

**Comment by erickpintor** [comment]
I'm getting a few errors while trying to test this branch:
- Running this into the repr: `q.Ref("indexes")`, returns: `q.Ref("undefined")`
- If I create a class, an index with no terms, and then try to navigate to it I get a `RangeError: Maximum call stack size exceeded.`

Do you think those are related to the depended PR you mentioned?

**Comment by sprsquish** [comment]
The second one sounds related. Not sure why the first one is doing that.

**Comment by sprsquish** [comment]
@erickpintor I'm not sure about either issue. I'm not seeing them when I try it

**Comment by erickpintor** [comment]
That's odd. I can't seem to simulate it anymore. Maybe I had some weird cache....
Anyway, are all tests passing for you? I have got some failures on this branch.

**Comment by sprsquish** [comment]
I think we still need the equality check from the JS driver released before everything will pass.

**Pull request merged**